docs/source/conf.py

```diff
@@ -287,3 +287,18 @@
 
 # Sort documentation in the same order as the source files.
 autodoc_member_order = 'bysource'
+
+
+#######################
+# Neutralize effects of function wrapping on documented signatures.
+# The affected signatures could be explcitly placed into the
+# documentation (either in .rst files or as a first line of a
+# docstring).
+import functools
+
+def no_op_wraps(func):
+    def wrapper(decorator):
+        return func
+    return wrapper
+
+functools.wraps = no_op_wraps
```

Comment:
Hmm.  This is probably rather unsafe.  It will only work as long as Sphinx itself doesn't depend on `functools.wraps` working properly - or if Sphinx manages to run all of its code that does depend on it before it evaluates this `conf.py`.

Both of these things might be true now but if that's the case this is still pretty fragile.  I'll try to make a follow-up suggestion for another way to address the problem this is aimed at (after I investigate a bit and see if there is another way).